File: exercism/haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples
  (sumOfMultiples)
  where

sumOfMultiples :: (Integral a) => [a] -> a -> a
sumOfMultiples fs n =
  sum . filter (\x -> 0 `elem` fmap (x `mod`) fs) $ [1 .. n - 1]

File: exercism/java/sum-of-multiples/src/main/java/SumOfMultiples.java
import java.util.stream.*;

class SumOfMultiples {

  int _s;

  SumOfMultiples(int number, int[] set) {
    _s =
        IntStream.range(0, number).filter(i -> IntStream.of(set).anyMatch(n -> 0 == (i % n))).sum();
  }

  int getSum() {
    return _s;
  }

}
